R/get_vimc_climate.R

```diff
@@ -0,0 +1,150 @@
+#' Get climate data Parquets prepared for VIMC
+#'
+#' @param country A string for the country name, or for the country ISO 2- or 3-
+#' character code. E.g. one of "Canada", "CA", or "CAN". Input is checked
+#' against names and codes in \pkg{countrycode}.
+#' @param date_range A two element vector of either strings or `<Date>`s, giving
+#' the interval for which to get data. If a character vector, must be in a
+#' format that can be converted to a `<Date>` by [lubridate::as_date()].
+#' @param data_location A location relative to the current working directory
+#' to search for the data. Data for `country` are expected to live in
+#' `data_location/XYZ` as Parquet files, where `XYZ` is the ISO 3-character code
+#' for `country`. Defaults to the working directory.
+#' @param data_source A string giving the climate data to search for. Names must
+#' match exactly.
+#' @param admin_level A numeric giving the GADM admin level for which to get
+#' data. Only one unit may be passed at a time.
+#'
+#' @return A `<data.frame>` of climate data.
+#'
+#' @export
+get_vimc_climate <- function(
+  country,
+  date_range,
+  data_location = here::here(),
+  data_source = c(
+    "CHIRPS",
+    "ERA5_mean",
+    "ERA5_min",
+    "ERA5_max",
+    "ERA5_RH",
+    "ERA5_SH",
+    "PERSIANN"
+  ),
+  admin_level = c(0, 1, 2, 3)
+) {
+  # more checks below
```

Comment:
Can you please consider using the existing utilities we have for this in reside.utils - we use them everywhere else and they are IMO much more compact than most of these checkmate-then-throw versions.  They're easy enough to expand if you need more.  

```
usethis::use_standalone("reside-ic/reside.utils", "utils-assert")
```

https://github.com/reside-ic/reside.utils/

That said, we don't have date handling in there (yet) I think but they'd be good to add - in the same style.  Perhaps start by writing similar assertions in this package, using lower-level ones there as required.